# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextEGL.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/gl/GLContextEGL.h
--- a/gfx/gl/GLContextEGL.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/gl/GLContextEGL.h	Wed Jun 10 22:21:02 2020 +0000
@@ -22,15 +22,13 @@
   friend class TextureImageEGL;
 
   static already_AddRefed<GLContextEGL> CreateGLContext(
-      GLLibraryEGL*, CreateContextFlags flags, const SurfaceCaps& caps,
-      bool isOffscreen, EGLConfig config, EGLSurface surface,
+      GLLibraryEGL*, const GLContextDesc&, EGLConfig config, EGLSurface surface,
       const bool useGles, nsACString* const out_failureId);
 
  public:
   MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(GLContextEGL, override)
-  GLContextEGL(GLLibraryEGL*, CreateContextFlags flags, const SurfaceCaps& caps,
-               bool isOffscreen, EGLConfig config, EGLSurface surface,
-               EGLContext context);
+  GLContextEGL(GLLibraryEGL*, const GLContextDesc&, EGLConfig config,
+               EGLSurface surface, EGLContext context);
 
   ~GLContextEGL();
 
@@ -91,11 +89,10 @@
   void Destroy();
 
   static already_AddRefed<GLContextEGL> CreateEGLPBufferOffscreenContext(
-      CreateContextFlags flags, const gfx::IntSize& size,
-      const SurfaceCaps& minCaps, nsACString* const out_FailureId);
+      const GLContextCreateDesc&, const gfx::IntSize& size,
+      nsACString* const out_FailureId);
   static already_AddRefed<GLContextEGL> CreateEGLPBufferOffscreenContextImpl(
-      CreateContextFlags flags, const gfx::IntSize& size,
-      const SurfaceCaps& minCaps, bool aUseGles,
+      const GLContextCreateDesc&, const gfx::IntSize& size, bool aUseGles,
       nsACString* const out_FailureId);
 
 #if defined(MOZ_WAYLAND) || defined(MOZ_WIDGET_ANDROID)